swarm/runtime/boundary_enforcement.py

```
@@ -411,23 +409,28 @@ def _check_secret_exposure(
-                                severity=ViolationSeverity.CRITICAL,
-                                path=file_path,
-                                operation="content_scan",
-                                detail=f"File {file_path} contains potential {desc} ('{pattern}...')",
-                                step_id=self._step_id,
-                                remediation="REMOVE THIS SECRET IMMEDIATELY. Use environment variables.",
-                            )
-                        )
-                        break  # Stop after first secret found in file
+                # Read text line by line to prevent memory exhaustion (ignore binary files)
+                secret_found = False
+                with open(abs_path, "r", encoding="utf-8") as f:
```

Ruff (0.15.9)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)